doc/arkode/guide/source/ARKodeSPRKStorage.rst

```diff
@@ -0,0 +1,177 @@
+.. ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2023, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SPRKStorage:
+
+==============================
+SPRK Method Storage Structure
+==============================
+
+.. _SPRKStorage.id:
+
+Method identifiers
+------------------
+
+The following enum values are used to identify different SPRK methods:
+
+.. c:macro:: ARKODE_SPRK_NONE
+
+   Identifier representing no SPRK method, this is solely used to mark the beginning of the enum.
+
+.. c:macro:: ARKODE_SPRK_EULER_1_1
+
+   Identifier for the Symplectic Euler 1st order method with 1 stage.
+
+.. c:macro:: ARKODE_SPRK_LEAPFROG_2_2
+
+   Identifier for the Symplectic Leapfrog 2nd order method with 2 stages.
+
+.. c:macro:: ARKODE_SPRK_PSEUDO_LEAPFROG_2_2
+
+   Identifier for the Symplectic Pseudo Leapfrog 2nd order method with 2 stages.
+
+.. c:macro:: ARKODE_SPRK_RUTH_3_3
+
+   Identifier for the Symplectic Ruth 3rd order method with 3 stages.
+
+.. c:macro:: ARKODE_SPRK_MCLACHLAN_2_2
+
+   Identifier for the Symplectic McLachlan 2nd order method with 2 stages.
+
+.. c:macro:: ARKODE_SPRK_MCLACHLAN_3_3
+
+   Identifier for the Symplectic McLachlan 3rd order method with 3 stages.
+
+.. c:macro:: ARKODE_SPRK_CANDY_ROZMUS_4_4
+
+   Identifier for the Symplectic Candy Rozmus 4th order method with 4 stages.
+
+.. c:macro:: ARKODE_SPRK_MCLACHLAN_4_4
+
+   Identifier for the Symplectic McLachlan 4th order method with 4 stages.
+
+.. c:macro:: ARKODE_SPRK_MCLACHLAN_5_6
+
+   Identifier for the Symplectic McLachlan 5th order method with 6 stages.
+
+.. c:macro:: ARKODE_SPRK_YOSHIDA_6_8
+
+   Identifier for the Symplectic Yoshida 6th order method with 8 stages.
+
+.. c:macro:: ARKODE_SPRK_SUZUKI_UMENO_8_16
+
+   Identifier for the Symplectic Suzuki-Umeno 8th order method with 16 stages.
+
+.. c:macro:: ARKODE_SPRK_SOFRONIOU_10_36
+
+   Identifier for the Symplectic Sofroniou 10th order method with 36 stages.
+
+.. c:type:: ARKodeSPRKStorage_s
```

Comment:
We could have `ARKodeSPRKStorageMem` as the private structure and `ARKodeSPRKStorage` as the public structure to mirror the Butcher table API. The reason I did not use `Table` is to differentiate it more from the Butcher table structure and with the forward thinking to other low storage methods, where it may be awkward to call the data structure a "Table"`.

Reply:
"Storage" seems a little generic to me. "Table" seemed more natural to me because the coefficients are from a Butcher table but stored an a specialized way to the SPRK case instead of as a generic Butcher table (the same would be true for a particular low storage pattern). We could (and maybe should) similarly have a `ARKodeIMEXTable` type to make it easier to load the tables for an IMEX method.